### ISSUE TITLE
iOS calls draw() 3x before setup()

### ISSUE THREAD

**Issue by kylemcdonald** [openframeworks/openFrameworks#5464]
I noticed I was getting a `texture has not been allocated` allocated and some other very confusing crashes. After stripping it way down I got to this point:

```c++
#include "ofMain.h"
#include "ofxiOS.h"

class ofApp : public ofxiOSApp {
public:
    void setup() {
        ofLog() << "setup";
    }
    void draw() {
        ofLog() << "draw";
    }
};

int main() {
    ofiOSWindowSettings settings;
    settings.enableRetina = true;
    settings.enableDepth = true;
    settings.enableAntiAliasing = true;
    settings.numOfAntiAliasingSamples = 4;
    settings.enableHardwareOrientation = true;
    settings.enableHardwareOrientationAnimation = false;
    settings.glesVersion = OFXIOS_RENDERER_ES1;
    settings.windowMode = OF_FULLSCREEN;
    ofCreateWindow(settings);
    return ofRunApp(new ofApp);
}
```

This code prints:

```
[notice ] draw
[notice ] draw
[notice ] draw
[notice ] setup
[notice ] draw
...
```

This behavior is consistent. I worked around this by adding a `bool ready = false;` near the top of ofApp, `ready = true;` at the end of `setup()` and `if(!ready) return;` at the top of `draw()`.

I couldn't find this documented anywhere, so apologies if this is the expected behavior on iOS :)

Running on an iPad Pro model ML0N2LL/A, iOS version 10.2.1, built with Xcode 8.2.1 and openFrameworks master (currently at 0c0a45dd6644c000c328302eb90bb7a4d52e2452 ). Project is set to build in debug mode (the default).

**Comment by kylemcdonald** [comment]
I also get this same behavior on an iPhone 6s (iOS version 10.2.1 built with XCode 8.2.1).

**Comment by jvcleave** [comment]
doesn't appear to be an issue in 0.9.8
Simple fix is to change:
https://github.com/openframeworks/openFrameworks/blob/master/addons/ofxiOS/src/core/ofxiOSViewController.mm#L63

`[self.glView performSelector:@selector(setup) withObject:nil afterDelay:0];`

to

`[self.glView setup];`

**Comment by arturoc** [comment]
this should be fixed now with #5918
